File: app/src/main/java/es/etologic/mahjongscoring2/app/old_games/OldGamesViewModelFactory.java
package es.etologic.mahjongscoring2.app.old_games;

import android.arch.lifecycle.ViewModel;
import android.support.annotation.NonNull;

import es.etologic.mahjongscoring2.app.base.BaseViewModelFactory;
import es.etologic.mahjongscoring2.domain.threading.UseCaseHandler;
import es.etologic.mahjongscoring2.domain.use_cases.DeleteGameUseCase;
import es.etologic.mahjongscoring2.domain.use_cases.GetGamesUseCase;

public class OldGamesViewModelFactory extends BaseViewModelFactory {

    private final GetGamesUseCase getOldGamesUseCase;
    private DeleteGameUseCase deleteGameUseCase;

    public OldGamesViewModelFactory(UseCaseHandler useCaseHandler,
                                    GetGamesUseCase getOldGamesUseCase,
                                    DeleteGameUseCase deleteGameUseCase) {
        super(useCaseHandler);
        this.getOldGamesUseCase = getOldGamesUseCase;
        this.deleteGameUseCase = deleteGameUseCase;
    }

    @SuppressWarnings ("unchecked")
    @NonNull
    @Override
    public <T extends ViewModel> T create(@NonNull Class<T> modelClass) {
        return (T)new OldGamesViewModel(useCaseHandler, getOldGamesUseCase, deleteGameUseCase);
    }
}

File: app/src/main/java/es/etologic/mahjongscoring2/app/main/old_games/OldGamesViewModelFactory.kt
package es.etologic.mahjongscoring2.app.main.old_games

import androidx.lifecycle.ViewModel
import androidx.lifecycle.ViewModelProvider
import es.etologic.mahjongscoring2.domain.use_cases.DeleteGameUseCase
import es.etologic.mahjongscoring2.domain.use_cases.GetGamesUseCase
import javax.inject.Inject

class OldGamesViewModelFactory @Inject
constructor(private val getGamesUseCase: GetGamesUseCase, private val deleteGameUseCase: DeleteGameUseCase) : ViewModelProvider.NewInstanceFactory() {
    
    override fun <T : ViewModel> create(modelClass: Class<T>): T {
        @Suppress("UNCHECKED_CAST")
        return OldGamesViewModel(getGamesUseCase, deleteGameUseCase) as T
    }
}
